source/types.ts

```diff
@@ -30,7 +30,9 @@ export interface NextcloudClientInterface extends NextcloudClientProperties {
   getFolderProperties(path: string, extraProperties?: FileDetailProperty[]):  Promise<FolderProperties>;
   configureWebdavConnection(options: ConnectionOptions):                      void;
   configureOcsConnection(options: ConnectionOptions):                         void;
-  pipeStream(path: string, stream: Stream.Readable):                          Promise<void>;
+  pipeStream(path: string, readStream: Stream.Readable):                      Promise<void>;
+  upload(targetPath: string, readStream: Stream.Readable):                    Promise<void>;
```

Comment:
What about `uploadFromStream` and `downloadToStream` ?
Or, to stay with `webdav` terms:

`postFromStream` and `getAsStream`

Reply:
uploadFromStream and downloadToStream sound ok